app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    database = {}
+
+    def inner(*args) -> Callable:
+        if args in database:
+            print("Getting from cache")
+            return database[args]
+        print("Calculating new result")
+        result = func(*args)
+        database[args] = result
+        return result
```

Comment:
it is better to use only one return here (cause u return the same thing here), for better code style

Reply:
Fixed.